test-runner/src/sys.rs

```diff
@@ -0,0 +1,142 @@
+use std::io;
+
+#[cfg(target_os = "macos")]
+pub fn reboot() -> Result<(), test_rpc::Error> {
+    unimplemented!("not implemented")
+}
+
+#[cfg(target_os = "windows")]
+pub fn reboot() -> Result<(), test_rpc::Error> {
+    use windows_sys::Win32::System::Shutdown::{
+        ExitWindowsEx, EWX_REBOOT, SHTDN_REASON_FLAG_PLANNED, SHTDN_REASON_MAJOR_APPLICATION,
+        SHTDN_REASON_MINOR_OTHER,
+    };
+    use windows_sys::Win32::UI::WindowsAndMessaging::EWX_FORCEIFHUNG;
+
+    grant_shutdown_privilege()?;
+
+    std::thread::spawn(|| {
+        std::thread::sleep(std::time::Duration::from_secs(5));
```

Comment:
Why do we need to sleep?

Reply:
This is just to prevent the test machine from shutting down before the test manager has received confirmation that it will shut down.